src/utilities/createRouter.ts

```diff
@@ -1,60 +1,68 @@
+import { readonly } from 'vue'
 import { Resolved, Route, RouteMethods, Routes } from '@/types'
-import { createRouteMethods, generateRouteRegexPattern, resolveRoutes, routeParamsAreValid } from '@/utilities'
+import { createRouteMethods, resolveRoutes } from '@/utilities'
+import { resolveRoutesRegex } from '@/utilities/resolveRoutesRegex'
+import { routeMatch } from '@/utilities/routeMatch'
+
+type RouterPush = (url: string, options?: { replace: boolean }) => Promise<void>
+type RouterReplace = (url: string) => Promise<void>
+type RouterNavigation = (number: number) => Promise<void>
 
 export type Router<
   TRoutes extends Routes
 > = {
   routes: RouteMethods<TRoutes>,
-  push: (url: string) => void,
-  replace: (url: string) => void,
-  back: () => void,
-  forward: () => void,
-  go: (number: number) => void,
-  routeMatch: (path: string) => Resolved<TRoutes[number]> | undefined,
+  route: Readonly<Resolved<Route>>,
```

Comment:
is this the "current route"? I expected that to be possibly undefined?

Reply:
Yes it's the current route. Which should never be undefined? When the app loads there should be a route right? 

Unless there's a rejection 🤔 is that what you're thinking?

Reply:
I guess my thinking was more in error states, but I don't know that it's worth having to check for undefined everywhere in the event that the user might be in one of those "computed routes" where something like not-found, or needs-auth is controlling the route. I didn't know if those were still going to be legit "routes"

Reply:
It's a great question that I hadn't thought of. I don't know the answer. Will have to think about that.